pkg/servicemesh/federation/server/routing_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/cluster"
+	"istio.io/istio/pkg/config/labels"
+	federationmodel "istio.io/istio/pkg/servicemesh/federation/model"
+)
+
+func TestCreateResourceName(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		mesh                 cluster.ID
+		source               federationmodel.ServiceKey
+		expectedResourceName string
+	}{
+		{
+			name: "standard mesh name + standard namespace + standard service name",
+			mesh: "mesh",
+			source: federationmodel.ServiceKey{
+				Namespace: "bookinfo",
+				Name:      "productpage",
+			},
+			expectedResourceName: "fed-exp-productpage-bookinfo-mesh",
+		},
+		{
+			name: "long mesh name",
+			mesh: "myveryveryverylongmesh",
+			source: federationmodel.ServiceKey{
+				Namespace: "bookinfo",
+				Name:      "productpage",
+			},
+			expectedResourceName: "fed-exp-productpage-bookinfo-myveryveryverylongmesh",
```

Comment:
Funny names : )
Should we test random string of a fixed length name cases ?

Reply:
I guess the main purpose of these tests is to have clear examples of how the function formats the resource name. Wouldn't adding and using a generated random string be to format the expected result `expectedResourceName` with it as the function would do ?

Reply:
ok, I see now. It's fine with those cases now.

Reply:
Could we log the length of a resource name if a test case fails ?
For example, name: "long mesh name", mesh: "myveryveryverylongmesh", if there is  anything failed, log the length of an input string name + mesh + ...